assets/js/base/components/block-error-boundary/index.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import { Component } from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import BlockError from './block-error';
+import './style.scss';
+
+class BlockErrorBoundary extends Component {
+	state = { hasError: false };
+
+	static getDerivedStateFromError( error ) {
+		return { errorMessage: error.message, hasError: true };
+	}
+
+	render() {
+		const { header, imageUrl, showErrorMessage, text } = this.props;
+		const { errorMessage, hasError } = this.state;
+
+		if ( hasError ) {
+			return (
+				<BlockError
+					header={ header }
+					imageUrl={ imageUrl }
+					errorMessage={ showErrorMessage ? errorMessage : null }
+					text={ text }
```

Comment:
I wonder if this prop should be a bit more descriptive or alternatively add some inline jsdocs to the component to make it clearer?

Reply:
Taking a second look at this, I replaced the `text` prop with `content`, so consumers of this component can not only change the text, but also the nodes rendered there (b9d045c295e9b3bfa3102824871817b80f5d6f56).

I also added docs to propTypes (65b84a252069d3595a486712f4e77db419f3d1b3).